dashboard/app/controllers/courses_controller.rb

```diff
@@ -43,7 +43,25 @@ def create
     end
   end
 
+  def update
+    course = Course.find_by_name!(params[:course_name])
+    course.update_and_persist(params[:scripts], i18n_params)
```

Comment:
That works in places like `level`, because the serialization is captured entirely in the level object. Course (and script) are weird in that in addition to serialization to `course_name.course`, we also need to save our localizeable data to a yml file.

We could save our localizeable data explicitly, and then have it do the serialization to `course_name.course` in an after_save.